code/modules/medical/pathology/pathogen_benevolent.dm

```diff
@@ -179,30 +179,37 @@ datum/pathogeneffects/benevolent/oxygenconversion
 	disease_act(var/mob/M as mob, var/datum/pathogen/origin)
 		if (!origin.symptomatic)
 			return
-		if (M:losebreath > 0)
-			M.TakeDamage("chest", M:losebreath * 2, 0)
-			M:losebreath = 0
-			if (prob(25))
-				M.show_message("<span class='alert'>You feel your body deteriorating as you breathe on.</span>")
-		if (M.get_oxygen_deprivation())
-			if (origin.stage != 0)
-				M.take_oxygen_deprivation(0 - (origin.stage / 2))
-
-datum/pathogeneffects/benevolent/oxygenproduction
-	name = "Oxygen Production"
-	desc = "The pathogen produces oxygen."
-	rarity = RARITY_VERY_RARE
+		var/mob/living/carbon/C = M
+		var/oxy_damage = max(20, C.get_oxygen_deprivation())
+		if (C.oxyloss || oxy_damage > 20)
+			var/datum/statusEffect/patho_oxy_speed/bad/POS = C.setStatus("patho_oxy_speed_bad", duration = INFINITE_STATUS)
+			POS.efficiency = origin.stage/2.5 // more efficient conversion at higher stages
```

Comment:
could set `efficiency` with the `optional` arg on on setStatus

Reply:
Thanks, I must have missed that the setStatus had an optional argument!